lib/DxilRootSignature/DxilRootSignature.cpp

```diff
@@ -190,6 +190,44 @@ void DeleteRootSignature(const DxilVersionedRootSignatureDesc * pRootSignature)
   delete pRootSignature;
 }
 
+template <typename RootSigDesc, typename RootParameterDesc>
+void ExtendRootSig(RootSigDesc &rootSigDesc, 
+                   uint32_t ShaderRegister,
+                   uint32_t RegisterSpace) {
+  auto *existingParams = rootSigDesc.pParameters;
+  auto *newParams = new RootParameterDesc[rootSigDesc.NumParameters + 1];
+  memcpy(newParams, existingParams, rootSigDesc.NumParameters * sizeof(RootParameterDesc));
```

Comment:
as above, I think this should be in the conditional.

Reply:
Yep